### PR TITLE
Feat/lw 9611 implement compute scriptdata hash function

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We need to implement in the SDK a function to compute the script data hash field of the transaction body, The script data hash can be computed as follows:

Format:
[ redeemers | datums | language views ]

The redeemers are exactly the data present in the transaction witness set. Similarly for the datums, if present. If no datums are provided, the middle field is an empty string.

Notice that script hashes are computed for the whole transaction, so if the transaction interacts with many scripts, all the redeemers and datums of all scripts must be included in the computation.

# Proposed Solution

Implement a `computeScriptDataHash` to compute the script data hash.

### CODE REVIEW

File: packages/tx-construction/src/computeScriptDataHash.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    /*
+     ; script data format:
+     ; [ redeemers | datums | language views ]
+     ; The redeemers are exactly the data present in the transaction witness set.
+     ; Similarly for the datums, if present. If no datums are provided, the middle
+     ; field is an empty string.
+    */
+    writer.writeEncodedValue(getCborEncodedArray(redemeers));
+
+    if (datums && datums.length > 0) writer.writeEncodedValue(getCborEncodedArray(datums));
```
Comment (iccicci) [inline packages/tx-construction/src/computeScriptDataHash.ts:72]: Reading the above comment I would have expected an `else` branch to write the empty string.

Could it be an empty string is represented as an empty buffer? Thank you

Comment (AngelCastilloB) [reply packages/tx-construction/src/computeScriptDataHash.ts:72]: Yes they mean by this literally '' (nothing)

File: packages/e2e/test/wallet/PersonalWallet/phase2validation.test.ts [hunk 2]
```diff
@@ -86,8 +97,15 @@ describe('PersonalWallet/phase2validation', () => {
       purpose: Cardano.RedeemerPurpose.mint
     };
 
-    // Script data hash was precomputed with the CML (hash of datums, redeemers and language views).
-    const scriptDataHash = Hash32ByteBase16('b6eb57092c330973b23784ac39426921eebd8376343409c03f613fa1a2017126');
+    // TODO: Currently cost models come from the backend out of order and as an object rather than an array.
```
[the remark this answers is not in the document]
Comment (AngelCastilloB) [reply packages/e2e/test/wallet/PersonalWallet/phase2validation.test.ts:100]: I think there must be, maybe @mirceahasegan can tell us

[the remark this answers is not in the document]
Comment (mirceahasegan) [reply packages/e2e/test/wallet/PersonalWallet/phase2validation.test.ts:100]: I would say it's LW-7691. Once we merge conway-era into master, we can get the cost models from the backend

[the remark this answers is not in the document]
Comment (AngelCastilloB) [reply packages/e2e/test/wallet/PersonalWallet/phase2validation.test.ts]: done and squashed

**Review by mkazlauskas — APPROVED** [review]
Nice work! :rocket:

**Review by mirceahasegan — APPROVED** [review]
Great work, @AngelCastilloB 💪

**Pull request merged**